Content.Server/_DV/CosmicCult/MonumentSystem.cs

```diff
@@ -0,0 +1,369 @@
+    [Dependency] private readonly DamageableSystem _damage = default!;
+    [Dependency] private readonly DamageableSystem _damageable = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly IConfigurationManager _config = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly IPrototypeManager _protoMan = default!;
+    [Dependency] private readonly ServerGlobalSoundSystem _sound = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedUserInterfaceSystem _ui = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<MonumentComponent, InteractUsingEvent>(OnInfuseHeldEntropy);
+        SubscribeLocalEvent<MonumentComponent, ActivateInWorldEvent>(OnInfuseEntropy);
+    }
+
+
+    public override void Update(float frameTime) // This Update() can fit so much functionality in it
+    {
+        base.Update(frameTime);
+
+        var finaleQuery = EntityQueryEnumerator<CosmicFinaleComponent, MonumentComponent>(); // Enumerator for The Monument's Finale
+        while (finaleQuery.MoveNext(out var uid, out var comp, out var monuComp))
+        {
+            if (_timing.CurTime >= monuComp.CheckTimer)
+            {
+                var entities = _lookup.GetEntitiesInRange(Transform(uid).Coordinates, 10);
+                entities.RemoveWhere(entity => !HasComp<InfluenceVitalityComponent>(entity));
+                foreach (var entity in entities) _damageable.TryChangeDamage(entity, monuComp.MonumentHealing * -1);
+                monuComp.CheckTimer = _timing.CurTime + monuComp.CheckWait;
+            }
+
+            if (comp.CurrentState == FinaleState.ActiveBuffer && _timing.CurTime >= comp.BufferTimer) // swap everything over when buffer timer runs out
+            {
+                comp.CurrentState = FinaleState.ActiveFinale;
+                comp.FinaleTimer = _timing.CurTime + comp.FinaleRemainingTime;
+                comp.SelectedSong = comp.FinaleMusic;
+
+                _sound.StopStationEventMusic(uid, StationEventMusicType.CosmicCult);
+                _appearance.SetData(uid, MonumentVisuals.FinaleReached, 3);
+                _chatSystem.DispatchStationAnnouncement(uid, Loc.GetString("cosmiccult-announce-finale-warning"), null, false, null, Color.FromHex("#cae8e8"));
+
+                Timer.Spawn(TimeSpan.FromSeconds(1),
+                    () =>
+                    {
+                        _sound.DispatchStationEventMusic(uid, comp.SelectedSong, StationEventMusicType.CosmicCult);
+                    });
+            }
+            else if (comp.CurrentState == FinaleState.ActiveFinale && _timing.CurTime >= comp.FinaleTimer) // trigger wincondition on time runout
```

Comment:
Explaining comments should be above or inside the relevant line or code block, not at the end of a line